pkg/api/v1/resource/helpers.go

```diff
@@ -91,12 +98,24 @@ func PodRequests(pod *v1.Pod, opts PodResourcesOptions) v1.ResourceList {
 			containerReqs = applyNonMissing(containerReqs, opts.NonMissingContainerRequests)
 		}
 
+		// Is the sidecar feature gate enabled and the init container marked as a sidecar?
+		if opts.SidecarContainersEnabled && container.RestartPolicy != nil && *container.RestartPolicy == v1.ContainerRestartPolicyAlways {
+			// and add them to the resulting cumulative container requests
+			addResourceList(reqs, containerReqs)
+
+			// track our cumulative sidecar resources
+			addResourceList(sidecarResources, containerReqs)
+			containerReqs = sidecarResources
+		}
+
 		if opts.ContainerFn != nil {
 			opts.ContainerFn(containerReqs, podutil.InitContainers)
 		}
-		maxResourceList(reqs, containerReqs)
+		maxResourceList(initContainerReqs, containerReqs)
```

Comment:
I think this should be more complicated. For `POD {Sidecar {limit 10} Init {limit200} regular {limit 100}}`. Total pod limit will be `210`. You need to add up all resources from sidecars before the given Init container and then take max of it

Reply:
Yup, too much context switching today, added your example as a test case and fixed my test case